compact_str/src/features/sqlx.rs

```diff
@@ -98,14 +99,17 @@ impl sqlx::postgres::PgHasArrayType for CompactString {
 #[cfg(feature = "sqlx-sqlite")]
 #[cfg_attr(docsrs, doc(cfg(feature = "sqlx-sqlite")))]
 impl<'q> Encode<'q, sqlx::Sqlite> for CompactString {
-    fn encode(self, buf: &mut <sqlx::Sqlite as HasArguments<'q>>::ArgumentBuffer) -> IsNull {
+    fn encode(
+        self,
+        buf: &mut <sqlx::Sqlite as Database>::ArgumentBuffer<'q>,
+    ) -> Result<IsNull, BoxDynError> {
         Encode::<'_, sqlx::Sqlite>::encode(self.into_string(), buf)
     }
 
     fn encode_by_ref(
         &self,
-        buf: &mut <sqlx::Sqlite as HasArguments<'q>>::ArgumentBuffer,
-    ) -> IsNull {
+        buf: &mut <sqlx::Sqlite as Database>::ArgumentBuffer<'q>,
+    ) -> Result<IsNull, BoxDynError> {
         Encode::<'_, sqlx::Sqlite>::encode(alloc::string::String::from(self.as_str()), buf)
```

Comment:
Similar to the above, here we have a [`SqliteArgumentValue`](https://docs.rs/sqlx/latest/sqlx/sqlite/enum.SqliteArgumentValue.html). Trying to use `self.as_str()` doesn't work because then the lifetime of `&self` would need to be longer than `'q`, but there's no way to enforce that because then our impl would contain more lifetimes than the definition of the `Encode` trait